final_task/rss_reader/rss_reader.py

```diff
@@ -0,0 +1,174 @@
+import requests
+import argparse
+import json
+import logging
+import socket
+import bs4
+from bs4 import BeautifulSoup
+
+
+log = logging.getLogger("Log")
+
+
+class CustomEx(Exception):
+    def __init__(self, msg):
+        self.msg = msg
+        super().__init__(msg)
+
+
+def is_connected() -> bool:
+    log.info("Checking internet connection...")
+    try:
+        socket.create_connection(("www.google.com", 80))
+        return True
+    except OSError:
+        pass
+    raise CustomEx("No internet connection")
```

Comment:
Try to rethink your sequence of actions here, this code can be simpler.
1. try to create connection
2. if anything is wrong - raise an exception
3. if everything ok - return true